cmd/template-resolver/utils/templateResolver.go

```diff
@@ -172,143 +198,261 @@ func ProcessTemplate(yamlBytes []byte, hubKubeConfigPath, clusterName string) ([
+	if err != nil {
+		return nil, fmt.Errorf("invalid JSON resulted after resolving templates: %w", err)
+	}
 
-			objectDefinition = policyTemplate["objectDefinition"].(map[string]interface{})
-			objectDefinitionUnstructured.Object = objectDefinition
+	resolvedYAML, err := templates.JSONToYAML(resolvedJSON)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert the processed Configuration Policy back to YAML: %w", err)
+	}
+
+	return resolvedYAML, nil
+}
+
+// ProcessObjTemplatesRaw takes a YAML string representation and
+// resolves the object's hub and managed templates
+func processObjTemplatesRaw(
+	raw unstructured.Unstructured,
+	resolver *templates.TemplateResolver,
+	hubResolver *templates.TemplateResolver,
+) ([]byte, error) {
+	if hubResolver != nil {
+		resolvedRaw, err := resolveHubTemplates(raw.Object, hubResolver)
+		if err != nil {
+			return nil, err
 		}
 
-		var rawDataList [][]byte
+		raw.Object = resolvedRaw
+	}
 
-		resolveOptions := templates.ResolveOptions{}
+	var rawDataList [][]byte
 
-		oTRaw, oTRawFound, _ := unstructured.NestedString(objectDefinition, "spec", "object-templates-raw")
-		if oTRawFound {
-			resolveOptions.InputIsYAML = true
+	resolveOptions := templates.ResolveOptions{}
 
-			rawDataList = [][]byte{[]byte(oTRaw)}
-		} else {
-			resolveOptions.InputIsYAML = false
+	oTRaw, oTRawFound, _ := unstructured.NestedString(raw.Object, "object-templates-raw")
+	if oTRawFound {
+		resolveOptions.InputIsYAML = true
 
-			objTemplates, _, err := unstructured.NestedSlice(objectDefinition, "spec", "object-templates")
-			if err != nil {
-				return nil, fmt.Errorf(
-					"invalid object-templates array in ConfigurationPolicy at policy-templates index %d: %w",
-					i,
-					err,
-				)
-			}
+		rawDataList = [][]byte{[]byte(oTRaw)}
+	} else {
+		return nil, fmt.Errorf("invalid object-templates-raw after resolving hub templates")
+	}
 
-			for _, objTemplate := range objTemplates {
-				jsonBytes, err := json.Marshal(objTemplate)
-				if err != nil {
-					return nil, fmt.Errorf(
-						"invalid object-templates array in ConfigurationPolicy at policy-templates index %d: %w",
-						i,
-						err,
-					)
-				}
-
-				rawDataList = append(rawDataList, jsonBytes)
-			}
+	objectTemplates, err := resolveObjTemplates(oTRawFound, rawDataList, resolver, resolveOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	unstructured.RemoveNestedField(raw.Object, "object-templates-raw")
+
+	err = unstructured.SetNestedSlice(raw.Object, objectTemplates, "object-templates")
+	if err != nil {
+		return nil, fmt.Errorf("failed to process the object-templates-raw: %w", err)
+	}
+
+	resolvedRaw, err := json.Marshal(raw.Object)
+	if err != nil {
+		return nil, fmt.Errorf("invalid JSON resulted after resolving templates: %w", err)
+	}
+
+	resolvedYAML, err := templates.JSONToYAML(resolvedRaw)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert the processed object-templates-raw back to YAML: %w", err)
+	}
```

Comment:
(Optional) This portion seems like a common step--it seems it could be moved back up into the `processTemplates()` function.